src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -371,5 +371,10 @@ public static void BuildDefaultConfigurationFile(StringBuilder builder, Network
 
             ConnectionManagerSettings.BuildDefaultConfigurationFile(builder, network);
         }
+
+        public void Dispose()
+        {
+            this.LoggerFactory?.Dispose();
```

Comment:
why `?`

Reply:
Also when you do this you should ensure that it is not this post elsewhere, which it is

Reply:
Still the question holds - Why the question mark?

Reply:
OK Can I do that then to ensure it is not set to null somewhere else in the class ?
![image](https://user-images.githubusercontent.com/4638821/43144477-280950dc-8f55-11e8-931e-5c1d8a7704bf.png)

Reply:
i'll check if it can be null

Reply:
https://github.com/monsieurleberre/StratisBitcoinFullNode/blob/cfba559d0f75d5abf4d302a4087939b4664b034e/src/Stratis.Bitcoin/Configuration/NodeSettings.cs#L93 - seems to be the only constructor and it sets up the factory, so it can NOT be null, we can remove `?`